internal/nginx/modules/httpmatches.js

```diff
@@ -0,0 +1,83 @@
+export default { redirect };
+
+const matchesVariable = 'http_matches';
+
+// FIXME(osborn): Need to add special handling for repeated headers.
+// Should follow guidance from https://www.rfc-editor.org/rfc/rfc7230.html#section-3.2.2.
+function headersMatch(r, headers) {
+  for (let i = 0; i < headers.length; i++) {
+    const h = headers[i];
+    const kv = h.split(':');
+    if (kv.length !== 2) {
+      r.error('invalid header match: ' + h);
+      return false;
+    }
+    let val = r.headersIn[kv[0]];
+    if (!val || val.toLowerCase() !== kv[1]) {
+      r.warn('header: ' + h + ' does not exist in request');
```

Comment:
is this warning really helpful? seems like it is normal case that a request doesn't include a header.
is it only for troubleshooting/debugging?

same comment for paramsMatch and a warning in line 60 about method.

Reply:
debug will probably be better. if there is a problem they can inspect, but normal operation is generally quiet.

it does make me think of a case for metrics; can we update some kind of blocked metric (eventually)?

Reply:
I wanted to use debug here but the request object `r` does not expose the debug log functionality. We can only log at the error, info, and warn levels.

Since these aren't really warning messages and we don't want to be too noisy, should I just remove these logs altogether?

Reply:
I went ahead and removed these logs.